flycheck.el

```diff
@@ -6068,6 +6069,33 @@ See URL `http://www.foodcritic.io'."
        ;; http://matschaffer.github.io/knife-solo/#label-Init+command
        (locate-dominating-file parent-dir "cookbooks")))))
 
+(flycheck-define-checker chicken-scheme
+  "A CHICKEN Scheme syntax checker using the CHICKEN compiler `csc'.
+
+See URL `http://call-cc.org/'."
+  :command ("csc" "-A" "-local" source)
+  :predicate
+  (lambda ()
+    (or (not (eq major-mode 'scheme-mode))
+        ;; In `scheme-mode' we must check the current Scheme implementation
+        ;; being used
+        (and (boundp 'geiser-impl--implementation)
+             (eq geiser-impl--implementation 'chicken))))
```

Comment:
@fice-t The Racket syntax checker supports two different major modes hence the extra check in the predicate.  But it doesn't make sense here.

@manuel-uberti Please remove the major mode check—it's never true since this syntax checker is only ever enabled for Scheme Mode.

Please also update the documentation of the syntax checker to state that Geiser Mode must be active for this syntax checker to work.